ts/components/InstabugButtons.tsx

```diff
@@ -19,25 +19,49 @@ type Props = ReturnType<typeof mapStateToProps> &
   OwnProps &
   ReturnType<typeof mapDispatchToProps>;
 
-class InstabugButtonsComponent extends React.PureComponent<Props, {}> {
+type State = {
+  instabugReportType: "bug" | "chat" | "undefined";
+};
+
+class InstabugButtonsComponent extends React.PureComponent<Props, State> {
   private handleIBChatPress = () => {
-    this.props.dispatchIBReportOpen("chat");
-    Chats.show();
+    this.setState({ instabugReportType: "chat" });
+    this.props.dispatchIBReportOpen(this.state.instabugReportType);
+    // Check if there are previous chat
+    Replies.hasChats(hasChats => {
+      if (hasChats) {
+        Replies.show();
+      } else {
+        Chats.show();
+      }
+    });
   };
 
   private handleIBBugPress = () => {
-    this.props.dispatchIBReportOpen("bug");
+    this.setState({ instabugReportType: "bug" });
+    this.props.dispatchIBReportOpen(this.state.instabugReportType);
     BugReporting.showWithOptions(BugReporting.reportType.bug, [
       BugReporting.option.commentFieldRequired
     ]);
   };
 
+  constructor(props: Props) {
+    super(props);
+    this.state = {
+      instabugReportType: "undefined"
+    };
+  }
   public componentDidMount() {
     // Register to the instabug dismiss event. (https://docs.instabug.com/docs/react-native-bug-reporting-event-handlers#section-after-dismissing-instabug)
     // This event is fired when chat or bug screen is dismissed
     BugReporting.onSDKDismissedHandler(
-      (dismissType: string, reportType: string): void => {
-        this.props.dispatchIBReportClosed(reportType, dismissType);
+      (dismiss: string): void => {
```

Comment:
```suggestion
      (dismiss: string, _: string): void => {
```
Leave a comment to explain why you are using the **instabugReportType** inside the state in place of report parameter